src/Tribe/Utils/Collection_Trait.php

```diff
@@ -152,14 +152,27 @@ public function count() {
 	 * {@inheritDoc}
 	 */
 	public function serialize() {
-		return serialize( $this->all() );
+		$to_serialize = $this->all();
+
+		if ( method_exists( $this, 'before_serialize' ) ) {
+			$to_serialize = $this->before_serialize( $this->all() );
+		}
+
+		return serialize( $to_serialize );
 	}
 
 	/**
 	 * {@inheritDoc}
 	 */
 	public function unserialize( $serialized ) {
-		$this->items = unserialize( $serialized );
+		$to_unserialize = $serialized;
+
+		if ( method_exists( $this, 'custom_unserialize' ) ) {
+			$this->items = $this->custom_unserialize( $to_unserialize );
+			return;
+		}
+
+		$this->items = unserialize( $to_unserialize );
```

Comment:
Got a two part question (...I'm unfamiliar with where these elements are going to be used, so this may not be a concern/may not be valid, but will ask anyway).

(1/2) For added safety on runtimes that support it, should we specify `allowed_classes` via `unserialize()`'s optional 2nd param? On early runtimes this won't do anything, but on more recent PHPs we'd be able to stop direct wakeup of arbitrary classes (which can be a security risk).

Reply:
The problem is that we cannot know the classes used in a general use collection implementation before-hand.

To keep track of those we would need to pass them at constructor time.
But this is a trait and we cannot control the `__construct` method. If special care should be taken to `unserialize`, then I think it should fall into classes using the trait, not the trait.